apollo-federation/src/operation/mod.rs

```diff
@@ -4878,10 +4879,15 @@ fn runtime_types_intersect(
     }
 
     if let (Ok(runtimes_1), Ok(runtimes_2)) = (
-        schema.possible_runtime_types(type1.clone()),
-        schema.possible_runtime_types(type2.clone()),
+        schema.possible_runtime_types_ref(type1.clone()),
+        schema.possible_runtime_types_ref(type2.clone()),
     ) {
-        return runtimes_1.intersection(&runtimes_2).next().is_some();
+        return match (runtimes_1, runtimes_2) {
+            (PossibleRuntimeTypes::Single(_), PossibleRuntimeTypes::Single(_)) => false,
```

Comment:
shouldn't this be `a == b`?

Reply:
in practice this is fine since

```
    if type1 == type2 {
        return true;
    }
```

is checked earlier in this function. But yeah ignore the uglyness of the implementation for now 😆